### PR TITLE
Basic working version of Stackdriver stats exporter

### PR DESCRIPTION
**Pull request by simonz130**
1) Fixed bugs related to Stackdriver Metrics Descriptor creation
2) Renamed Metrics Exporter to Stats Exporter to follow Opencensus terminology
3) Cache Stackdriver Metrics Descriptors locally so it's easy which descriptor holds the time series to upload
4) Fixed test code that was producing the metrics using Opencensus and uploading it to Stackdriver
5) Minor refactoring

### CODE REVIEW

File: src/OpenCensus.Exporter.Stackdriver/Implementation/GoogleCloudResourceUtils.cs [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+            resource.Labels.Add(Constants.PROJECT_ID_LABEL_KEY, projectId);
+
+            // TODO - zeltser - setting monitored resource labels for detected resource
+            // along with all the other metadata
+
+            return resource;
+        }
+
+        /*
+        public static Resource detectResource()
```
Comment (SergeyKanzhelev) [inline src/OpenCensus.Exporter.Stackdriver/Implementation/GoogleCloudResourceUtils.cs]: do you need commented code in this repo?

File: src/Samples/TestStackdriver.cs [hunk 2]
```diff
@@ -28,7 +28,8 @@ internal class TestStackdriver
             VideoSizeViewName,
             "processed video size over time",
             VideoSize,
-            Distribution.Create(BucketBoundaries.Create(new List<double>() { 0.0, 16.0 * MiB, 256.0 * MiB })),
+            Sum.Create(),
+            //Distribution.Create(BucketBoundaries.Create(new List<double>() { 0.0, 16.0 * MiB, 256.0 * MiB })),
```
Comment (SergeyKanzhelev) [inline src/Samples/TestStackdriver.cs]: Why is this commented?

Comment (simonz130) [reply src/Samples/TestStackdriver.cs]: I used a different aggregation for testing. Removing the comment.

File: src/OpenCensus.Exporter.Stackdriver/Implementation/GoogleCloudResourceUtils.cs (hunk 1 above)
Comment (simonz130) [reply src/OpenCensus.Exporter.Stackdriver/Implementation/GoogleCloudResourceUtils.cs]: No, fixing it.

**Comment by simonz130** [comment]
This is now ready to be merged.

I'm on UTC+2 timezone until the end of this week, so plan to add support for environment detection later on (today/tomorrow) as a separate PR, building on this one.

**Comment by SergeyKanzhelev** [comment]
Please update the branch. If I'd update it - CLA automation may not like it

**Comment by simonz130** [comment]
Can you try merging now?

**Pull request merged**